updates to shared memory tests

RemiSoummer opened the pull request:
Following #308 I was doing some tests so thought I might as well have a quick PR to add add a bit more testing here. 
Not sure how useful that will be here.

RemiSoummer committed:
Update to the existing test

RemiSoummer committed:
adding a couple basic tests

ehpor commented:
I'd use `pytest.raises(RuntimeError, match='string to match to the exception string')` instead of an assert after the context manager. See https://docs.pytest.org/en/stable/reference/reference.html#pytest-raises

RemiSoummer committed:
updated to use match and typo fix

RemiSoummer commented:
@ehpor also I was trying to be fancy can catch the specific error message, but that is not working on Windows.  So I reverted that to the base message from `SharedMemory.cpp`. 
Also, for the `TypeError` I don't have the full string so `match`  does not work, and for now I reverted to the previous syntax. 
If you have another suggestion I can implement, otherwise we can skip that test if you prefer. 

Unfortunately there is no `contains=` option in `pytest.raises`

ehpor commented:
@RemiSoummer `match` is a full-on regex string. So it matches substrings in the exception string. An example from the docs:

```python
with pytest.raises(ValueError, match='must be 0 or None'):
    raise ValueError("value must be 0 or None")
```

RemiSoummer committed:
removing the test for empty-string shared memory which is allowed on Windows

RemiSoummer commented:
Ok thanks - for some reason I must have done the wrong thing and though match was not working.

RemiSoummer committed:
reverting to partial string in match to have specific error message

RemiSoummer commented:
ha but not that doesn't work on windows, so can't do here, back to filtering on generic message

RemiSoummer committed:
Revert "reverting to partial string in match to have specific error message"

This reverts commit d25b654dce289a79fdf9bfab27fb27726dd20519.

ehpor reviewed (approved):
LGTM. And you fixed my typo in line 9 as well XD

FYI: your hardship is the reason why people usually don't bother testing for specific error messages. It makes the tests fragile.
